tensorflow/core/kernels/mkl/mkl_dequantize_op.cc

```diff
@@ -53,15 +55,16 @@ class MklDequantizeOp : public OpKernel {
       auto cpu_engine = engine(engine::kind::cpu, 0);
 
       // Get the inputs
-      const Tensor& src_tensor = MklGetInput(ctx, kSrcIndex);
-      const float min_range =
-          MklGetInput(ctx, kMinIndex).template flat<float>()(0);
-      const float max_range =
-          MklGetInput(ctx, kMaxIndex).template flat<float>()(0);
+      const Tensor& src_tensor = ctx->input(kSrcIndex);
+      const float min_range = ctx->input(1).flat<float>()(0);
+      const float max_range = ctx->input(2).flat<float>()(0);
 
       // Get MklShape
       MklDnnShape src_mkl_shape;
       GetMklShape(ctx, kSrcIndex, &src_mkl_shape, native_format);
+      auto src_tf_shape = src_mkl_shape.IsMklTensor()
+                              ? src_mkl_shape.GetTfShape()
+                              : src_tensor.shape();
```

Comment:
You can assume input is always NonMklTensor (just use shape() function). Other existing usages of IsMklTensor() will be cleaned up later.